### ISSUE TITLE
Update lvSingleDefRegCandidate after CSE to get more candidates for EHwrite thru

### ISSUE THREAD

**Issue by kunalspathak** [dotnet/runtime#86200]
https://github.com/dotnet/runtime/issues/66578#issuecomment-1071752551

> While working on this issue, I realized that since lvSingleDefRegCandidate is set during lvaMarkLclRefs (which is very earlier phase), lot of single defs which are produced in CSE are missed out and we do not enregister them in EHWrite thru (single-def) scenarios. I will look into it in a separate PR.

**Comment by AndyAyersMS** [comment]
A while back I added code to put some new CSE temps into SSA, probably similar criteria apply for is single def (note CSEs can have multiple defs, so not all will qualify).

https://github.com/dotnet/coreclr/pull/26986

**Comment by kunalspathak** [comment]
I tried this prototype in https://github.com/kunalspathak/runtime/commit/5e093f8ec4a073b8c62199370a08227df93f3b0b.

- First, I moved the marking of `singleDef` from early phase to the late phase, but that doesn't work because currently, throughput the phases, we mark the variables as `doNotEnregister`. When we have to mark a varDsc as `singleDef`, we won't do that if the `VarDsc` is already marked with `doNotEnregister`. With that, we don't see many variables getting marked as `single-def` and we lose the ability of enregistering the EH vars.
- Next, I tried to keep the marking of `singleDef` in two places. One, that we already have currently and second, during the liveness update after lowering. At this point, I need to only check variables that were created after the first time we marked variables as `singleDef`. Mostly they are CSE created variables. However, most of them are not EH var (and are very local) and hence marking them as `singleDef` doesn't change much.
- As a cleanup, I also tried to remove existing code of marking `singleDef` and replace it with the information we have during LSRA (`interval->isSingleDef`), but relying on that is too late because we have already identified the candidates until that time. So we need to know before LSRA about `singleDef` information.

I don't think there is anything actionable at this point.
